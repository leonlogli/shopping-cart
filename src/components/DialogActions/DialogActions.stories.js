import React from 'react'

import DialogActions from '.'

export default {
  title: 'DialogActions',
}

export const withButtonText = () => (
  <DialogActions open okText="Save" cancelText="Abort" />
)

export const withChildren = () => <DialogActions>custom actions</DialogActions>
