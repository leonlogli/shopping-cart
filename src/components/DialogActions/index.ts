export * from './DialogActions'
export { default } from './DialogActions'
