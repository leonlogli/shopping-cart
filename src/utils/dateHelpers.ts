const durationFormat = (interval: number, type: string) => {
  return `il y a ${Math.floor(interval)} ${type}${
    Math.floor(interval) > 1 ? 's' : ''
  }`
}

const fromNow = (date: Date) => {
  const seconds = Math.floor(
    ((new Date() as any) - (new Date(date) as any)) / 1000
  )

  let interval = seconds / 31536000

  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'an')
  }

  interval = seconds / 2592000
  if (Math.floor(interval) >= 1) {
    return `il y a ${Math.floor(interval)} mois`
  }

  interval = seconds / 86400
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'jour')
  }

  interval = seconds / 3600
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'heure')
  }

  interval = seconds / 60
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'minute')
  }

  return `Ã  l'instant`
}

/**
 * Parse the given string time to a date.Time must have `hh:mm:ss` format
 * @param time time to parse
 * @param date date to which the time will be set once parsed. If null, today's date is used
 */
const parseTimeToDate = (time: string, date?: Date) => {
  const dateToSet = date || new Date()
  const timeParts = time.split(':').map((t) => Number(t) || 0)

  dateToSet.setHours(timeParts[0], timeParts[1], timeParts[2] || 0)

  return dateToSet
}

const formatShortDate = (date: any, locale?: string) => {
  return new Date(date).toLocaleDateString(locale, {
    day: 'numeric',
    month: 'short',
    year: 'numeric',
  })
}

export { fromNow, parseTimeToDate, formatShortDate }
