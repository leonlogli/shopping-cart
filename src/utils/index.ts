export * from './testHelpers'
export * from './util'
export * from './dateHelpers'
