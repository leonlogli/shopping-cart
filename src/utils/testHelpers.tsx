import React from 'react'
import { ThemeProvider } from 'styled-components'
import { mount, shallow } from 'enzyme'
import { defaultTheme } from '../styles'

const ThemeProviderWrapper = ({ children }: any) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
)

export const shallowWithTheme = (tree: any) =>
  shallow(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })

export const mountWithTheme = (tree: any) =>
  mount(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })
