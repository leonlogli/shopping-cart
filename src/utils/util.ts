const toUSD = (value: number) => {
  return `${value.toLocaleString('en-US', {
    minimumFractionDigits: 2,
  })} USD`
}

export { toUSD }
