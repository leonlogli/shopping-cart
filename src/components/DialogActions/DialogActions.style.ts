import styled from 'styled-components'
import DialogActions from '@material-ui/core/DialogActions'

const DialogActionsRoot = styled(DialogActions)`
  padding-left: 16px;
  padding-right: 16px;
`

export { DialogActionsRoot }
export default DialogActionsRoot
