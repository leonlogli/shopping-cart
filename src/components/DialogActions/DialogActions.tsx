import React, { MouseEvent } from 'react'
import { DialogActionsProps as MuiDialogActionsProps } from '@material-ui/core/DialogActions'
import Button, { ButtonProps } from '@material-ui/core/Button'

import DialogActionsRoot from './DialogActions.style'

export type DialogActionsProps = MuiDialogActionsProps & {
  /**
   * The text of the Cancel button	string
   */
  cancelText?: string
  /**
   * The text of the Confirm button
   */
  okText?: string
  /**
   * A callback of ok button
   */
  onOk?: (event: MouseEvent<HTMLButtonElement>) => void
  okButtonProps?: Omit<ButtonProps, 'onClick'>
  cancelButtonProps?: Omit<ButtonProps, 'onClick'>
  /**
   * A callback of cancel button
   */
  onCancel?: (event: MouseEvent<HTMLButtonElement>) => void
}

const DialogActions = (props: DialogActionsProps) => {
  const {
    children,
    onOk,
    cancelText,
    cancelButtonProps,
    okText,
    onCancel,
    okButtonProps,
  } = props

  return (
    <DialogActionsRoot>
      {children}
      {cancelText && (
        <Button color="primary" {...cancelButtonProps} onClick={onCancel}>
          {cancelText}
        </Button>
      )}
      {okText && (
        <Button color="primary" {...okButtonProps} onClick={onOk}>
          {okText}
        </Button>
      )}
    </DialogActionsRoot>
  )
}

export default DialogActions
export { DialogActions }
