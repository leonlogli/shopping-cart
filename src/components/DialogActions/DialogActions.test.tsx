import React from 'react'

import { mountWithTheme } from '../../utils'
import DialogActions from '.'

describe('<DialogActions />', () => {
  const wrapper = mountWithTheme(
    <DialogActions okText="Save" cancelText="Cancel" />
  )

  it('should render children', () => {
    expect(wrapper.find('button').length).toEqual(2)
  })

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should set button text', () => {
    wrapper.setProps({ okText: 'Ok' } as any)

    expect(wrapper.find('button').last().text()).toBe('Ok')
  })
})
