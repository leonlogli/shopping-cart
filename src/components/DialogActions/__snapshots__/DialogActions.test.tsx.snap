// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogActions /> should match a snapshot 1`] = `
<DialogActions
  cancelText="Cancel"
  okText="Save"
>
  <Styled(WithStyles(ForwardRef(DialogActions)))>
    <WithStyles(ForwardRef(DialogActions))
      className="dCGWfN"
    >
      <ForwardRef(DialogActions)
        className="dCGWfN"
        classes={
          Object {
            "root": "MuiDialogActions-root",
            "spacing": "MuiDialogActions-spacing",
          }
        }
      >
        <div
          className="MuiDialogActions-root dCGWfN MuiDialogActions-spacing"
        >
          <WithStyles(ForwardRef(Button))
            color="primary"
          >
            <ForwardRef(Button)
              classes={
                Object {
                  "colorInherit": "MuiButton-colorInherit",
                  "contained": "MuiButton-contained",
                  "containedPrimary": "MuiButton-containedPrimary",
                  "containedSecondary": "MuiButton-containedSecondary",
                  "containedSizeLarge": "MuiButton-containedSizeLarge",
                  "containedSizeSmall": "MuiButton-containedSizeSmall",
                  "disableElevation": "MuiButton-disableElevation",
                  "disabled": "Mui-disabled",
                  "endIcon": "MuiButton-endIcon",
                  "focusVisible": "Mui-focusVisible",
                  "fullWidth": "MuiButton-fullWidth",
                  "iconSizeLarge": "MuiButton-iconSizeLarge",
                  "iconSizeMedium": "MuiButton-iconSizeMedium",
                  "iconSizeSmall": "MuiButton-iconSizeSmall",
                  "label": "MuiButton-label",
                  "outlined": "MuiButton-outlined",
                  "outlinedPrimary": "MuiButton-outlinedPrimary",
                  "outlinedSecondary": "MuiButton-outlinedSecondary",
                  "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                  "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                  "root": "MuiButton-root",
                  "sizeLarge": "MuiButton-sizeLarge",
                  "sizeSmall": "MuiButton-sizeSmall",
                  "startIcon": "MuiButton-startIcon",
                  "text": "MuiButton-text",
                  "textPrimary": "MuiButton-textPrimary",
                  "textSecondary": "MuiButton-textSecondary",
                  "textSizeLarge": "MuiButton-textSizeLarge",
                  "textSizeSmall": "MuiButton-textSizeSmall",
                }
              }
              color="primary"
            >
              <WithStyles(ForwardRef(ButtonBase))
                className="MuiButton-root MuiButton-text MuiButton-textPrimary"
                component="button"
                disabled={false}
                focusRipple={true}
                focusVisibleClassName="Mui-focusVisible"
                type="button"
              >
                <ForwardRef(ButtonBase)
                  className="MuiButton-root MuiButton-text MuiButton-textPrimary"
                  classes={
                    Object {
                      "disabled": "Mui-disabled",
                      "focusVisible": "Mui-focusVisible",
                      "root": "MuiButtonBase-root",
                    }
                  }
                  component="button"
                  disabled={false}
                  focusRipple={true}
                  focusVisibleClassName="Mui-focusVisible"
                  type="button"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
                    disabled={false}
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Cancel
                    </span>
                    <WithStyles(memo)
                      center={false}
                    >
                      <ForwardRef(TouchRipple)
                        center={false}
                        classes={
                          Object {
                            "child": "MuiTouchRipple-child",
                            "childLeaving": "MuiTouchRipple-childLeaving",
                            "childPulsate": "MuiTouchRipple-childPulsate",
                            "ripple": "MuiTouchRipple-ripple",
                            "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                            "rippleVisible": "MuiTouchRipple-rippleVisible",
                            "root": "MuiTouchRipple-root",
                          }
                        }
                      >
                        <span
                          className="MuiTouchRipple-root"
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            component={null}
                            exit={true}
                          />
                        </span>
                      </ForwardRef(TouchRipple)>
                    </WithStyles(memo)>
                  </button>
                </ForwardRef(ButtonBase)>
              </WithStyles(ForwardRef(ButtonBase))>
            </ForwardRef(Button)>
          </WithStyles(ForwardRef(Button))>
          <WithStyles(ForwardRef(Button))
            color="primary"
          >
            <ForwardRef(Button)
              classes={
                Object {
                  "colorInherit": "MuiButton-colorInherit",
                  "contained": "MuiButton-contained",
                  "containedPrimary": "MuiButton-containedPrimary",
                  "containedSecondary": "MuiButton-containedSecondary",
                  "containedSizeLarge": "MuiButton-containedSizeLarge",
                  "containedSizeSmall": "MuiButton-containedSizeSmall",
                  "disableElevation": "MuiButton-disableElevation",
                  "disabled": "Mui-disabled",
                  "endIcon": "MuiButton-endIcon",
                  "focusVisible": "Mui-focusVisible",
                  "fullWidth": "MuiButton-fullWidth",
                  "iconSizeLarge": "MuiButton-iconSizeLarge",
                  "iconSizeMedium": "MuiButton-iconSizeMedium",
                  "iconSizeSmall": "MuiButton-iconSizeSmall",
                  "label": "MuiButton-label",
                  "outlined": "MuiButton-outlined",
                  "outlinedPrimary": "MuiButton-outlinedPrimary",
                  "outlinedSecondary": "MuiButton-outlinedSecondary",
                  "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                  "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                  "root": "MuiButton-root",
                  "sizeLarge": "MuiButton-sizeLarge",
                  "sizeSmall": "MuiButton-sizeSmall",
                  "startIcon": "MuiButton-startIcon",
                  "text": "MuiButton-text",
                  "textPrimary": "MuiButton-textPrimary",
                  "textSecondary": "MuiButton-textSecondary",
                  "textSizeLarge": "MuiButton-textSizeLarge",
                  "textSizeSmall": "MuiButton-textSizeSmall",
                }
              }
              color="primary"
            >
              <WithStyles(ForwardRef(ButtonBase))
                className="MuiButton-root MuiButton-text MuiButton-textPrimary"
                component="button"
                disabled={false}
                focusRipple={true}
                focusVisibleClassName="Mui-focusVisible"
                type="button"
              >
                <ForwardRef(ButtonBase)
                  className="MuiButton-root MuiButton-text MuiButton-textPrimary"
                  classes={
                    Object {
                      "disabled": "Mui-disabled",
                      "focusVisible": "Mui-focusVisible",
                      "root": "MuiButtonBase-root",
                    }
                  }
                  component="button"
                  disabled={false}
                  focusRipple={true}
                  focusVisibleClassName="Mui-focusVisible"
                  type="button"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
                    disabled={false}
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Save
                    </span>
                    <WithStyles(memo)
                      center={false}
                    >
                      <ForwardRef(TouchRipple)
                        center={false}
                        classes={
                          Object {
                            "child": "MuiTouchRipple-child",
                            "childLeaving": "MuiTouchRipple-childLeaving",
                            "childPulsate": "MuiTouchRipple-childPulsate",
                            "ripple": "MuiTouchRipple-ripple",
                            "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                            "rippleVisible": "MuiTouchRipple-rippleVisible",
                            "root": "MuiTouchRipple-root",
                          }
                        }
                      >
                        <span
                          className="MuiTouchRipple-root"
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            component={null}
                            exit={true}
                          />
                        </span>
                      </ForwardRef(TouchRipple)>
                    </WithStyles(memo)>
                  </button>
                </ForwardRef(ButtonBase)>
              </WithStyles(ForwardRef(ButtonBase))>
            </ForwardRef(Button)>
          </WithStyles(ForwardRef(Button))>
        </div>
      </ForwardRef(DialogActions)>
    </WithStyles(ForwardRef(DialogActions))>
  </Styled(WithStyles(ForwardRef(DialogActions)))>
</DialogActions>
`;
